internal/ask/form/form.go

```diff
@@ -102,6 +102,8 @@ func Upsert(context interface{}, db *db.DB, form *Form) error {
 	if form.ID.Hex() == "" {
 		form.ID = bson.NewObjectId()
 		isNewForm = true
+		form.DateCreated = time.Now()
+		form.DateUpdated = time.Now()
 	}
```

Comment:
It should still be added at that point, as it's important to keep the database as consistent as possible. If UpdateStats were to fail, then the DateUpdated would never be changed for a non-new form.